### DISCUSSION TITLE
How to configure renovate to update the latest release branch?

### DISCUSSION THREAD

**Discussion by ylevchenko1** [renovatebot/renovate#23355] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

1 Update a specific dependency for a specific branch.
2 if possible.
   Are there any possible configs/option to configure renovate to define the latest branch?
   e.g. I have branches release/3.2.2 and release/3.2.1, renovate should update the latest branch

### What you tried so far.

I have several question regarding renovate config
1 I have this config, it works as expected, it updates main, release/x.x.x, 3.3.2 branches for all kinds of dependencies
```
{
"$schema": "https://docs.renovatebot.com/renovate-schema.json",
"extends": ["config:base"],
"automerge": true,
"automergeType": "branch",
"prCreation": "not-pending",
"prHourlyLimit": 100,
"prConcurrentLimit" : 10,
"requiredStatusChecks": null,
"platformAutomerge": true,
"internalChecksAsSuccess": true,
"baseBranches": ["main", "/^release\/[3-9]{1}\\.[3-9]{1}\\.[2-9]{1}/", "3.3.2"]
}
```
regex **/^release\/[3-9]{1}\\.[3-9]{1}\\.[2-9]{1}/**  matches release/3.3.2 and higher releases
However if I replace baseBranches on packageRules, in this case renovate ignore all branches except main branch.
```
"packageRules": [
     {"matchBaseBranches":["/^release\/[3-9]{1}\\.[3-9]{1}\\.[2-9]{1}/", "3.3.2"],
     "matchPackagePatterns": ["org.mockito"]}
                 ]
}
```
Can matchBaseBranches  be used to update specific dependencies s for specific branches?
2 Are there any ways(configs) to only update the latest release branch or the only option to set a certain branch are  (matchBaseBranches, baseBranches  via exact branchName matching and regex)? e.g. I have release/3.2.1 and release/3.2.2 branches, renovate should update only the  release/3.2.2 branch (edited) 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
I don't understand your description. But renovate will update every dependency it finds in every base branch you define.

**Reply by ylevchenko1** [reply]
e.g. 
I have next branches  main, release/3.2.1, release/3.2.2
Can I somehow configure **baseBranches** in a such way(in main branch)
`"baseBranches": ["main", "release/<latest_version>"]`

to not configure the latest release branch every time after an every new release, I mean
`"baseBranches": ["main", "release/3.2.2"]`
after a new release
`"baseBranches": ["main", "release/3.2.3"]`
after the next new release
`"baseBranches": ["main", "release/3.2.4"]`

**Reply by rarkins** [reply, marked as answer]
No, that's not supported. If you're interested in implementing it yourself and have some ideas on how it could fit into renovate's existing base branch model then we can advise you

**Reply by viceice** [reply]
base branches supports regex

**Reply by rarkins** [reply]
Yes but not limiting to only one match or how to sort them

**Reply by viceice** [reply]
yes, that's right
